spring-xd-dirt/src/main/java/org/springframework/xd/dirt/plugins/spark/streaming/SparkStreamingMessageConverterSupport.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.xd.dirt.plugins.spark.streaming;
+
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.integration.channel.AbstractMessageChannel;
+import org.springframework.util.MimeType;
+import org.springframework.xd.dirt.plugins.stream.ModuleTypeConversionPlugin;
+import org.springframework.xd.dirt.plugins.stream.ModuleTypeConversionSupport;
+import org.springframework.xd.dirt.util.ConfigLocations;
+
+/**
+ * Utility class to support message conversion in spark streaming module.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+public class SparkStreamingMessageConverterSupport {
```

Comment:
This is a nice idea. I see some package constraints with this approach. The `SparkMessageSender` (which is in **spring-xd-spark-streaming** and subsequently in spring-xd-module) needs to extend this `SparkStreamingChannel` so that it could be used to send messages in `ModuleExecutor` while the SparkStreamingChannel could only be in **spring-xd-dirt** package as it needs ModuleTypeConversionPlugin